helix-core/src/main/java/org/apache/helix/controller/rebalancer/waged/WagedRebalancer.java

```diff
@@ -69,8 +68,8 @@ private ResourceChangeDetector getChangeDetector() {
   public WagedRebalancer(HelixManager helixManager) {
     // TODO init the metadata store according to their requirement when integrate, or change to final static method if possible.
     _assignmentMetadataStore = new AssignmentMetadataStore();
-    // TODO init the algorithm according to the requirement when integrate.
-    _rebalanceAlgorithm = new ConstraintsRebalanceAlgorithm();
+    // TODO parse the cluster setting
```

Comment:
Question, will it be parsed inside the Factory? That seems the reasonable solution from my point of view.

Reply:
We can make it that way later, the rebalancer class doesn't have ClusterConfig object yet. We can think about how it's constructed during the integration phase.